scripts/split-e2e-tests-codebuild.ts

```diff
@@ -0,0 +1,272 @@
+import * as glob from 'glob';
+import * as fs from 'fs-extra';
+import { join } from 'path';
+import * as yaml from 'js-yaml';
+import { migrationFromV10Tests, migrationFromV5Tests, migrationFromV6Tests } from './split-e2e-test-filters';
+// Ensure to update packages/amplify-e2e-tests/src/cleanup-e2e-resources.ts is also updated this gets updated
+const AWS_REGIONS_TO_RUN_TESTS = [
+  'us-east-1',
+  'us-east-2',
+  'us-west-2',
+  'eu-west-2',
+  'eu-central-1',
+  'ap-northeast-1',
+  'ap-southeast-1',
+  'ap-southeast-2',
+];
+// Some services (eg. amazon lex, containers) are not available in all regions
+// Tests added to this list will always run in the specified region
+const FORCE_REGION_MAP = {
+  'interactions': 'us-west-2',
+  'containers': 'us-east-1',
+}
+type FORCE_TESTS = 'interactions' | 'containers';
+// some tests require additional time, the parent account can handle longer tests (up to 90 minutes)
+const USE_PARENT_ACCOUNT = [
+  'api-key-migration2',
+  'api-key-migration3',
+  'api-key-migration4',
+  'api-key-migration5',
+  'searchable-migration',
+  'FunctionTransformerTestsV2'
+];
+const REPO_ROOT = join(__dirname, '..');
+const TEST_TIMINGS_PATH = join(REPO_ROOT, 'scripts', 'cci', 'test-timings.data.json');
+const CODEBUILD_CONFIG_BASE_PATH = join(REPO_ROOT, 'codebuild_specs', 'e2e_workflow_base.yml');
+const CODEBUILD_GENERATE_CONFIG_PATH = join(REPO_ROOT, 'codebuild_specs', 'e2e_workflow.yml');
+const RUN_SOLO = [
+  'src/__tests__/containers-api-1.test.ts',
+  'src/__tests__/containers-api-2.test.ts',
+  'src/__tests__/schema-auth-11.test.ts',
+  'src/__tests__/schema-auth-15.test.ts',
+  'src/__tests__/schema-connection.test.ts',
+  'src/__tests__/transformer-migrations/searchable-migration.test.ts',
+];
+const EXCLUDE_E2E_TESTS = [
```

Comment:
Currently filter out the test running longer than 1 hour. Will re-add once the assuming role issue is fixed.